### PR TITLE
Add test to check numeric precision GH33234

### LINKED ISSUE #33234: Groupby multiple columns causes agg to have precision loss on int64
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

df=pd.DataFrame({
    "KEY1": ["KEY1"],
    "KEY2": ["KEY2"],
    "INT64": [1583715738627261039]
})

# Grouping by one column produces correct output
print(df.groupby(["KEY1"]).agg(lambda x: x))

# Grouping by more than one column produces incorrect output
print(df.groupby(["KEY1", "KEY2"]).agg(lambda x: x))
```
#### Problem description

Precision loss on the int64 column when grouping by multiple columns.

Similar behaviour can be seen when doing
```
>>> df.groupby(["KEY1"]).sum()
                    INT64
KEY1                     
KEY1  1583715738627260928
```
but that seems like a different issue with cythonized group sum not supporting int64? According to: https://github.com/pandas-dev/pandas/issues/15027#issuecomment-269902341

#### Expected Output
```
      KEY2                INT64
KEY1                           
KEY1  KEY2  1583715738627261039
                         INT64
KEY1 KEY2                     
KEY1 KEY2  1583715738627261039
```
#### Actual Output
```
      KEY2                INT64
KEY1                           
KEY1  KEY2  1583715738627261039
                         INT64
KEY1 KEY2                     
KEY1 KEY2  1583715738627260928
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.6.5.final.0
python-bits      : 64
OS               : Linux
OS-release       : 3.16.0-77-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_AU.UTF-8
LOCALE           : en_AU.UTF-8

pandas           : 1.0.3
numpy            : 1.18.2
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 46.1.3
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by liang3zy22**
This a new PR to fix #33234 since old PR closed.

### CODE REVIEW

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 1]
```diff
@@ -1367,6 +1367,43 @@ def func(ser):
...
+            "key2": ["b"],
+            "key3": [
+                1583715738627261039
+                if dtype not in ["int32", "uint32", "Int32", "UInt32"]
+                else 4294967295
+                if dtype in ["uint32", "UInt32"]
+                else 2147483647
+            ],
+        }
+    ).astype({"key3": dtype})
```
Comment (phofl) [inline pandas/tests/groupby/aggregate/test_aggregate.py:1401]: Can you define all those objects explicitly? e.g. no astype

Comment (liang3zy22) [reply pandas/tests/groupby/aggregate/test_aggregate.py:1401]: Ok. Could I  drop the astype part and keep just DataFrame creation expression?

Comment (liang3zy22) [reply pandas/tests/groupby/aggregate/test_aggregate.py]: I think I defined all those objects explicitly in the updated commit.

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 2]
```diff
@@ -1377,6 +1377,43 @@ def func(ser):
...
+        "float32",
+        "Int64",
+        "UInt64",
+        "Int32",
+        "UInt32",
+        "Float64",
+        "Float32",
+    ],
+)
+def test_groupby_agg_precision(dtype):
```
Comment (mroeschke) [inline pandas/tests/groupby/aggregate/test_aggregate.py:1397]: Could you use the `any_real_numeric_dtype` fixture here?

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 3]
```diff
@@ -1377,6 +1377,43 @@ def func(ser):
...
+    # GH33234
+    df = DataFrame(
+        {
+            "key1": ["a"],
+            "key2": ["b"],
+            "key3": [
+                pd.array([1583715738627261039], dtype=dtype)[0]
```
Comment (mroeschke) [inline pandas/tests/groupby/aggregate/test_aggregate.py:1404]: Could you specify a `max_value` variable above this `df` assignment and then write this as

```
df = pd.DataFrame({..., "key3": pd.array([max_value], dtype=dtype)})
```

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 4]
```diff
@@ -1377,6 +1377,43 @@ def func(ser):
...
+                if dtype not in ["int32", "uint32", "Int32", "UInt32"]
+                else pd.array([4294967295], dtype=dtype)[0]
+                if dtype in ["uint32", "UInt32"]
+                else pd.array([2147483647], dtype=dtype)[0]
+            ],
+        }
+    )
+    expected = df[["key3"]]
+    result = df.groupby(["key1", "key2"]).agg(lambda x: x).reset_index()[["key3"]]
```
Comment (mroeschke) [inline pandas/tests/groupby/aggregate/test_aggregate.py:1413]: Could you construct `expected` such that `result` is just `df.groupby(["key1", "key2"]).agg(lambda x: x)`?

File: pandas/tests/groupby/aggregate/test_aggregate.py (hunk 2 above)
Comment (liang3zy22) [reply pandas/tests/groupby/aggregate/test_aggregate.py]: Done. Just only used `any_real_numeric_dtype `.

File: pandas/tests/groupby/aggregate/test_aggregate.py (hunk 4 above)
Comment (liang3zy22) [reply pandas/tests/groupby/aggregate/test_aggregate.py]: Done

**Review by mroeschke — APPROVED** [review]

**Pull request merged**